doc/Programming-guide/Key-platform-features/Endpoint-ownership/index.md

```diff
@@ -167,24 +166,422 @@ The configuration should match the following Avro schema.
 }
 ```
 
-### Administration ###
-To create a Twitter user verifier, use either [Admin UI]() or [REST API](). The following REST API call example illustrates how to create a new Twitter user verifier.
+##### Administration #####
+To create a Twitter owner verifier, use either Administration UI or [REST API]({{root_url}}Programming-guide/Server-REST-APIs/#TODO).
+The following image example illustrates how to create a Twitter owner verifier from Administration UI.
+                                   
+![twitter verifier](Admin-ui/verifier-twitter.png "twitter verifier")
+
+#### Custom owner verifier ####
+It is possible to implement and plug-in custom owner verifiers. You can find corresponding instructions on the 
+[Creating custom owner verifier]({{root_url}}Customization-guide/Kaa-Server/Development-environment-setup/#creating-custom-user-verifier) page.
+
+
+## Endpoint access token flow ##
+In the *endpoint access token flow*, new endpoints are attached to the owner with the help of the endpoint which was attached to the owner beforehand.
+The following steps illustrate this flow with the endpoint A, which is already attached to the owner, and the endpoint B, which is due to be attached.
+
+   1. The endpoint B periodically generates and sends its access token to the Kaa cluster.
+   2. The endpoint B displays its access token as a QR code on the screen (TV) or on the webpage (e.g., a router or other device with an embedded server).
+   3. The endpoint A retrieves this token by scanning QR code or in any other suitable way and sends it in the endpoint attach request to the Kaa cluster.
+   4. The Kaa cluster verifies the access code and attaches the endpoint B to the owner of the endpoint A.
+
+
+![Owner access token flow](endpoint_attach_2.png "Owner access token flow")
+
+
+### Attach endpoint to owner
+
+To enable sending/receiving events to/from endpoints, at first the client should attach the endpoint to the owner as shown in the following screenshot.
+
+> Please refer to [Glossary]({{root_url}}Glossary/kaa-admin) for more detail about **userExternalId**, **userAccessToken**, **userVerifierToken** 
+>and other parameters which used by methods of KaaClient.
```

Comment:
... for more detail<b>s</b> ... which <b>are</b> used by KaaClient methods ~~of KaaClient~~.